### ISSUE TITLE
The room is open to everyone but not allowed

### ISSUE THREAD

**Issue by Wokennn** [haxball/haxball-issues#1446]
TypeError: Cannot read property 'includes' of null
    at initPlayerStats (<anonymous>:95:19)
    at room.onPlayerJoin (<anonymous>:225:2)
    at u.rh (headless-min.js:64)
    at R.apply (headless-min.js:109)
    at Fa.nf (headless-min.js:45)
    at Fa.ec (headless-min.js:46)
    at A.lh (headless-min.js:63)
    at Function.Ea.va (headless-min.js:53)
    at headless-min.js:49

the room is open to everyone but unable to login it gives this error how can i solve this problem?

**Comment by Raamyy** [comment]
@Wokennn We don't have a magic wizard that we can use to view your code unfortunately :(.

Please post sample of the code from `room.onPlayerJoin` function and `initPlayerStats` function so anybody can help :smile:.

**Comment by Wokennn** [comment]
![IMG_20210825_105238](https://user-images.githubusercontent.com/89461489/130750015-ebf9708b-c9da-482a-ba17-3bad8fbfd516.jpg)

**Comment by genals** [comment]
The error is obviously in the function `initPlayerStats`, line 95, where includes() is used to access a string or array that has the value null (which makes it neither an array nor a string and causes the error).

@Raamyy Now we have to ask the `magic wizard` how to solve the error :)

**Comment by Wokennn** [comment]
![IMG_20210825_130230](https://user-images.githubusercontent.com/89461489/130771012-4b29f4e0-bba4-4268-891d-ad6ac56e76f8.jpg)

@genals is the mistake here? How should I do here?

**Comment by genals** [comment]
> ![IMG_20210825_130230](https://user-images.githubusercontent.com/89461489/130771012-4b29f4e0-bba4-4268-891d-ad6ac56e76f8.jpg)
> 
> @genals is the mistake here? How should I do here?

Yes the mistake is there. You are looking in kayitliNames for the string 'player.name' and not for the name of the player. Also kayitliNames seems to be null.

**Comment by Wokennn** [comment]
@genals so what should i do? I don't know exactly how to fix.

**Comment by genals** [comment]
> @genals so what should i do? I don't know exactly how to fix.

It is not easy to help you if you only post screenshots of your code and parts of it are in Turkish.

One cannot see where kayitliNames is defined and what else is happening with it in the code.

You have to debug your code. For example, write the value of kayitliNames in the console and see what changes and when.

Your problem has nothing to do with HaxBall per se. You just have to learn how to program. Such mistakes happen every now and then but usually they are not difficult to solve.

And a little hint at the end: use [Array.push()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) to append elements to the end of an array.

**Comment by Raamyy** [comment]
@Wokennn  Although I have no idea what that `kayitliNames` means, but I don't think that you want to look for `'player.name'` inside of it you should look for `player.name` as a variable not a string.

As @genals said I also suggest for you to look for some programing/javascript tutorials that will help you solve your problems.

**Comment by ghost** [comment]
> @Wokennn Although I have no idea what that `kayitliNames` means, but I don't think that you want to look for `'player.name'` inside of it you should look for `player.name` as a variable not a string.
> 
> As @genals said I also suggest for you to look for some programing/javascript tutorials that will help you solve your problems.

kayitliNames means registeredNames
